src/Bicep.Core/Analyzers/Linter/LinterRule.cs

```diff
@@ -0,0 +1,144 @@
+        {
+
+            this.ConfigHelper = new ConfigHelper();
+            this.AnalyzerName = LinterAnalyzer.AnalyzerName;
+            this.Code = code;
+            this.RuleName = ruleName;
+            this.Description = description;
+            this.DocumentationUri = docUri;
+
+            this.EnabledForCLI = enableForCLI;
+            this.EnabledForEditing = enableForEdit;
+            this.DiagnosticLevel = diagnosticLevel;
+
+            LoadConfiguration();
+        }
+
+        private readonly ConfigHelper ConfigHelper;
+        public string AnalyzerName { get; }
+        public string Code { get; }
+        public string RuleName { get; }
+
+        public const string RuleConfigSection = "Linter:Rules";
+
+        // TODO: Decide how we want to manage configuration
+        // Variants:
+        //  1. enable globally or onchange or onsave (i.e. static whole file)
+        //  2. set severity of diagnostic
+        //  3. when fix is present what about auto application
+
+        public bool EnabledForEditing { get; private set; }
+        public bool EnabledForCLI { get; private set; }
+        public Diagnostics.DiagnosticLevel DiagnosticLevel { get; private set; }
+        public string Description { get; }
+        public string DocumentationUri { get; }
+
+        private void LoadConfiguration()
+        {
+            this.EnabledForCLI = GetConfiguration(nameof(this.EnabledForCLI), this.EnabledForCLI);
+            this.EnabledForEditing = GetConfiguration(nameof(this.EnabledForEditing), this.EnabledForEditing);
+
+            var configDiagLevel = GetConfiguration(nameof(this.DiagnosticLevel), this.DiagnosticLevel.ToString());
+            if(DiagnosticLevel.TryParse<DiagnosticLevel>(configDiagLevel, true, out var lvl)){
+                this.DiagnosticLevel = lvl;
+            }
+        }
+
+        /// <summary>
+        /// GetMessage allows a linter rule display message to be dynamic without
+        /// resorting to side-effect inducing work in the Description property.
+        /// Should be overridden in any rule with a complex message requirement.
+        /// </summary>
+        /// <returns></returns>
+        protected virtual string GetMessage() => this.Description;
+
+        /// <summary>
+        /// Gets a formatted message using the supplied parameter values.
+        /// In the base class this ignores the parameters and will throw
+        /// an exception if called in Debug build.
+        /// </summary>
+        /// <param name="values"></param>
+        /// <returns></returns>
+        protected virtual string GetFormattedMessage(params object[] values)
+        {
+            Debug.Assert(values == null || values.Length == 0, "LinterRule GetFormattedMessage when needed should always be overridden. Values are ignored in base class.");
+            return this.Description;
+        }
+
+        public abstract IEnumerable<IBicepAnalyzerDiagnostic> Analyze(SemanticModel model);
```

Comment:
>SemanticModel [](http://example.com/codeflow?start=70&length=13)

Do we need to expose the entire `Compilation` to support cross-file linter rules?